Compiler.Lexer/Compiler.Lexer/Tokenizer.cs

```diff
@@ -0,0 +1,403 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Compiler.Lexer
+{
+    public partial class Tokenizer
+    {
+        private CharStream Stream;
+        private Token LastToken;
+
+        public Tokenizer(string source)
+        {
+            Stream = new CharStream(source);
+            LastToken = new Token
+                (TokenKind.UnknownToken, string.Empty, 0, 0);
+        }
+
+        public Token Get()
+        {
+            Token get;
+            int lineNumber = Stream.CurrentLineNumber;
+            int position = Stream.CurrentPosition;
+
+            do
+            {
+                if (Stream.EOF)
+                {
+                    return null;
+                }
+            } while (IsWhiteSpace());
+
+            get = IsSymbol();
+            if (get != null)
+            {
+                return get;
+            }
```

Comment:
you can have a factory here that returns a tokenizer which in turn returns a token. There can be a Tokenizer base class and sub classed to SymbolTokenize, OperatorTokenizer, etc

so you can have something like 
var internalTokenizer = tokenizerFactory.GetFrom(Stream);
return internalTokenizer.Get();
This improves reuse and readability substantially.